pkgs/crypto/keys/client/sign.go

```diff
@@ -26,48 +31,73 @@ var DefaultSignOptions = SignOptions{
 }
 
 func signApp(cmd *command.Command, args []string, iopts interface{}) error {
-	var kb keys.Keybase
-	var err error
 	var opts SignOptions = iopts.(SignOptions)
+	var err error
 
 	if len(args) != 1 {
 		cmd.ErrPrintfln("Usage: sign <keyname>")
 		return errors.New("invalid args")
 	}
-	if opts.AccountNumber == nil {
-		return errors.New("invalid account number")
-	}
-	if opts.Sequence == nil {
-		return errors.New("invalid sequence")
-	}
-
-	name := args[0]
-	txpath := opts.TxPath
-	kb, err = keys.NewKeyBaseFromDir(opts.Home)
-	if err != nil {
-		return err
-	}
+	opts.Name = args[0]
 
 	// read tx to sign
-	var tx std.Tx
-	var txjson []byte
+	txpath := opts.TxPath
 	if txpath == "-" { // from stdin.
 		txjsonstr, err := cmd.GetString("Enter tx to sign, terminated by a newline.")
 		if err != nil {
 			return err
 		}
-		txjson = []byte(txjsonstr)
+		opts.TxJson = []byte(txjsonstr)
 	} else { // from file
-		txjson, err = ioutil.ReadFile(txpath)
+		opts.TxJson, err = ioutil.ReadFile(txpath)
 		if err != nil {
 			return err
 		}
 	}
-	err = amino.UnmarshalJSON(txjson, &tx)
+
+	if opts.Quiet {
+		opts.Pass, err = cmd.GetPassword("")
+	} else {
+		opts.Pass, err = cmd.GetPassword("Enter password.")
+	}
 	if err != nil {
 		return err
 	}
 
+	signedJson, err := SignHandler(opts)
+	if err != nil {
+		return err
+	}
+
+	cmd.Println(string(signedJson))
+
+	return nil
+}
+
+func SignHandler(opts SignOptions) (string, error) {
```

Comment:
Implementation separated from the CLI parsing + exported, so it can be called from the cod/